### ISSUE TITLE
Notice of Trial Issued

### ISSUE THREAD

**Issue by mariahkannenberg** [flexion/ef-cms#1193]
As the Court, in order to notify the parties that their case has been set for trial, a Notice of Trial is issued about 5 months in advance of the start date of the trial session.  

## Acceptance Criteria
* When a calendar is set/automatically scheduled, all cases on that session should be issued a "Notice of Trial" (format in comments)
* If a case is manually added to a trial session *before* it is set/automatically scheduled, the case should not be issued a issued a "Notice of Trial" until that automatic scheduling
* If a case is manually added to a trial session *after* it is set/automatically scheduled, that case should be issued a "Notice of Trial" at the time the case is added to the trial session
* User should be able to easily see the sessions for which notices have already been issued (UX feedback; 9/2019 visit)
* When the Notice of Trial is issued, it is added to the Docket Record as a separate entry and served on the parties:
   * served date added to document; 
   * docket record displays served date and parties; 
   * parties receiving electronic service get email; 
   * parties receiving paper service get paper.
* Code for Docket Record = NDT
* Full address and Judge are required fields before a case is auto-scheduled (calendared)

## Notes
* This story does not include the issuance of a "Standing Pre-Trial Order" (for regular cases) or a "Standing Pre-Trial Notice" (for small cases) -- that is handled in #3910
*required fields to set trial Address 1 city state and zip
- 150 cases in a trial

## Tasks
_Tasked in trello_

## Definition of Done (Updated 8-28-19)
**Product Owner**
 - [x]  Acceptance criteria have been met

**UX**
 - [x] Business test scenarios to meet all acceptance criteria have been written
 - [x] Usability has been validated
 - [x] Wiki has been updated (if applicable) 
 - [x] Story has been tested on a mobile device (for external users only)

**Engineering**
- [x] Automated test scripts have been written (Matt)
- [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html (William)
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required) (All Screens Internal)
 - [x] Module dependencies are up-to-date and are at the latest resolvable version (npm update) (Joseph)
 - [x] Errors in Sonarcloud are fixed https://sonarcloud.io/organizations/flexion-github/projects (Joseph)
 - [x] Lambdas include CloudWatch logging of users, inputs and outputs
 - [x] Code refactored for clarity and to remove any known technical debt 
 - [x] Deployed to the dev environment
 - [x] Deployed to the stage environment

**Comment by JessicaMarine** [comment]
[Notice of Trial - Proposed.pdf](https://github.com/flexion/ef-cms/files/3941287/Notice.of.Trial.-.Proposed.pdf)

**Comment by alsmith1** [comment]
![Screen Shot 2019-12-13 at 10.12.22 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/69a5b2b9-982c-4a51-815a-3f58a6a71436)![Screen Shot 2019-12-13 at 10.21.05 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/74adfaa5-e623-48a0-890a-e83649ecb374)![Screen Shot 2019-12-13 at 10.21.18 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/e2bffc11-2b8c-4da2-9a23-14229fd3eab5)![Screen Shot 2019-12-13 at 10.21.30 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/2ace7b8e-bb99-4312-82d7-e6c02c917819)![Screen Shot 2019-12-13 at 10.21.38 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/f175f4cd-d4a9-4463-bd78-eadae047ce86)![Screen Shot 2019-12-13 at 10.21.44 AM.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/a9ca195b-6d68-4b6b-8b94-f5394c07c778)

**Comment by alsmith1** [comment]
***DRAFT***
—DRAFT—
## Test scenarios
#### Scenario 1 - When a calendar is set/automatically scheduled, all cases on that session should be issued a "Notice of Trial"
1. Login as any internal user
2. Navigate to trial sessions and select a session
3. Verify this session has not been calendared, the set calendar button appears
4. Scroll down to the eligible cases section and click on the Set Calendar Button
5. Verify a modal displays” are you sure you want to set the calendar for this trial session?”
6. Click Yes, set calendar
7. Verify success message displays at top of session information screen
8. Click on a case’s docket number in the trial session
9. Verify that a notice of set trial has been added to the docket record
10. Verify that the pdf in the document displays with the following
    1. Date  under court address
    2. Header : Notification Setting Case For Trial
    3. Case Caption, Petitioner and docket number info
    4. Trial information and judge information in titled boxes
    5. Text of letter with dynamically set [time] and [date]
    6. Clerk of court signature

Expected result
- [ ] Session has not been calendared and displays set calendar button
- [ ] Modal displays unpin clicking of set calendar button “are you sure you want to set the calendar for this trial session?”
- [ ] Upon clicking yes on modal a success message displays “eligible cases have been set for this trial”
- [ ] Notice of set trial displays in Case’s docket record
- [ ] The notice pdf in the document displays with the following
    1. Date  under court address
    2. Header : Notification Setting Case For Trial
    3. Case Caption, Petitioner and docket number info
    4. Trial information and judge information in titled boxes
    5. Text of letter with dynamically set [time] and [date]
    6. Clerk of court signature

#### Scenario 2 - If a case is manually added to a trial session before it is set/automatically scheduled, the case should not be issued a issued a "Notice of Trial" until that automatic scheduling
1. Login as any internal user
2. Locate a case that is ready for trial
3. Schedule the case above to an upcoming trial session that has not been calendared yet
4. Navigate to the docket record and Verify no notice is generated

Expected result
- [ ] No notice of set trial displays in the docket record 


#### Scenario 3 - If a case is manually added to a trial session after it is set/automatically scheduled, that case should be issued a "Notice of Trial" at the time the case is added to the trial session
1. Login as any internal user
2. Locate a case that is ready for trial
3. Schedule the case above to an upcoming trial session that has been calendared
4. Verify on the trial session selected displays a column Notice issued
5. Navigate to the docket record and Verify a notice of set trial is generated

Expected result
- [ ] Column displays on trial sessions page “notice issued” 
- [ ] Date displays under notice issued column for selected 
- [ ] Notice of set trial displays in the docket record

#### Scenario 4 - User should be able to easily see the sessions for which notices have already been issued
1. Login as any internal user
2. Navigate to the trial sessions page
3. Click on one of the upcoming trials and click set calendar
4. Verify on the trial session page, the selected trial now displays a column Notice issued
5. Verify in the Notice issued column for selected trial session a date displays on when notice was issued

Expected result
- [ ] On the trial session page, the selected trial now displays a column Notice issued
- [ ] Notice issued column for selected trial session displays a date on when notice was issued

#### Scenario 5 - When the Notice of Trial is issued, it is added to the Docket Record as a separate entry and served on the parties
1. Login as any internal user
2. Locate a case that is ready for trial
3. Schedule the case above to an upcoming trial session that has been calendared
4. Navigate to the docket record and Verify a notice of set trial is generated

Expected result
- [ ] Notice of set trial displays in the docket record


#### Scenario 6 - Full address and Judge are required fields before a case is auto-scheduled (calendared)
1. Login as any internal user
2. Navigate to the trial sessions page
3. Verify a trial that does not have full court address or judge listed
4. Click set calendar
5. Verify a yellow warning message displays at the top of the session information page. “This trial session requires additional information” “You must provide an address and judge to be able to set this trial session” displays

Expected result
- [ ] A trial displays that does not have full court address or judge listed 
- [ ] a yellow warning message displays at the top of the session information page. “This trial session requires additional information” “You must provide an address and judge to be able to set this trial session” displays

**Comment by alsmith1** [comment]
Location of address for notice
![741a3752-f589-44b2-8f95-82794c0fd22b.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/0a0f1856-90d5-46b0-943e-a2cf1ed0b92f)![cbefee35-5662-47c3-a7c9-ca9abf882f35.png](https://images.zenhubusercontent.com/5d766a771251c6000116623d/c86abb51-e415-4b78-8dea-11c77aed502a)

**Comment by Tinabuck5** [comment]
Scenario 1:  I created 152-20 and 153-20 Honolulu, HI.  Everything appears to work properly.   Also, we should run by Jessica on the docket record the notice of trial says filed by Barbara Wooten.  We usually see filed by a Petitioner, a third party or respondent.  When serving notices that are created by the Court the "filed by" is usually blank.  We should let Jessica weigh in on that.

Scenario 2:  Works as expected.

Scenario 3:  Works as expected

Scenario 4:  Works as expected

Scenario 5: Works as expected

Scenario 6:  Works as expected

Scenario 7:  I attempted to set a Small Fresno session.  I could not proceed without completing the information.  A yellow warning screen box displayed.  
I ran into a few other quirky things but will send in a separate message.
